src/plugins/explore/public/application/utils/hooks/use_page_initialization.ts

```diff
@@ -0,0 +1,103 @@
+import { useSavedExplore } from './use_saved_explore';
+import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
+import { ExploreServices } from '../../../types';
+import {
+  setSavedSearch,
+  setChartType,
+  setStyleOptions,
+  setAxesMapping,
+  setQueryState,
+  setActiveTab,
+  clearResults,
+  clearQueryStatusMap,
+  clearLastExecutedData,
+  setEditorMode,
+} from '../state_management/slices';
+import { executeQueries } from '../state_management/actions/query_actions';
+import { ExploreFlavor } from '../../../../common';
+import { useSetEditorText } from '../../hooks';
+import { EditorMode } from '../state_management/types';
+
+export const useInitPage = () => {
+  const dispatch = useDispatch();
+  const { services } = useOpenSearchDashboards<ExploreServices>();
+  const exploreId = useCurrentExploreId();
+  const { savedExplore, error } = useSavedExplore(exploreId);
+  const setEditorText = useSetEditorText();
+  const { chrome, data } = services;
+
+  useEffect(() => {
+    if (savedExplore && !error) {
+      if (savedExplore.id) {
+        // Deserialize state from saved object
+        const { title } = savedExplore;
+
+        // Update browser title and breadcrumbs
+        chrome.docTitle.change(title);
+        chrome.setBreadcrumbs([{ text: 'Explore', href: '#/' }, { text: title }]);
+
+        // Sync query from saved object to data plugin (explore doesn't use filters)
+        const searchSourceFields = savedExplore.kibanaSavedObjectMeta;
+        if (searchSourceFields?.searchSourceJSON) {
+          const searchSource = JSON.parse(searchSourceFields.searchSourceJSON);
+          const query = searchSource.query;
+          if (query) {
+            dispatch(setQueryState(query));
+            setEditorText(query.query);
+          }
+        }
+
+        // Update savedSearch to store just the ID (like discover)
+        // TODO: remove this once legacy state is not consumed any more
+        dispatch(setSavedSearch(savedExplore.id));
+
+        // Init vis state and ui state
+        const visualization = savedExplore.visualization;
+        const uiState = savedExplore.uiState;
+        if (visualization) {
+          const { chartType, params, axesMapping } = JSON.parse(visualization);
+          dispatch(setChartType(chartType));
+          dispatch(setStyleOptions(params));
+          dispatch(setAxesMapping(axesMapping));
+        }
+        if (uiState) {
+          const { activeTab } = JSON.parse(uiState);
+          dispatch(setActiveTab(activeTab));
+        }
+
+        // Add to recently accessed
+        chrome.recentlyAccessed.add(
+          `/app/explore/${savedExplore.type ?? ExploreFlavor.Logs}#/view/${savedExplore.id}`,
+          title,
+          savedExplore.id,
+          { type: 'explore' }
+        );
+
+        dispatch(clearLastExecutedData());
+        dispatch(setEditorMode(EditorMode.Query));
+        dispatch(clearResults());
```

Comment:
i notice we kind of done a lot of of this? was there way to have a single dispatch event that does this. because i think i had to dispatch all of these and becomes a little bit not maintainable. would be easier to dispatch a single event and then who ever consumed this handled this